### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by dalehamel**
@dwradcliffe @andrewjamesbrown for review

This is my initial attempt at a hack to make test kitchen work with our roles.

I had to inject my logic partway through the inheritance chain, which is definitely hacky because I had to call _before_ the sandbox is created, but I want to steal all of the logic from the chef zero provisioner.

If any create_sandbox in the inheritance chain changes, the methods here will need to be updated to reflect that. 

For safety, we might want to staple a 'compatible test-kitchen version' into here, and bail if the test-kitchen version being used doesn't match.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: ruby
+rvm:
+- 2.0.0
+- 1.9.3
```
Comment (dwradcliffe) [inline .travis.yml]: 1.9.3 is really old

File: LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Copyright (c) 2013 David Genord II
```
Comment (dwradcliffe) [inline LICENSE.txt]: wat?

**Comment by dwradcliffe** [comment]
Makes sense

**Comment by andrewjamesbrown** [comment]
This is super complicated for what it's actually doing.  Going to submit a new PR that simplifies things...

**Pull request merged**